client/js/postManager.js

```diff
@@ -0,0 +1,56 @@
+class PostManager {
+        
+    constructor(myUrl) {
+        this.url = myUrl;
+    }
+    
+    postData(inUrl, inData) {
+        
+        var outData = undefined;
+        
+        $.ajax({
+            url: this.url+"/"+inUrl,
+            type: "POST",
+            data: JSON.stringify(inData),
+            contentType: "application/json",
+            dataType: "json",
+            success: function(data, status){
+                if (status != "success") {
+                    console.log("data not sent due to " + status);
+                } else {
+                    outData = data;
+                }
+            }
+        });
+        
+        return outData;
```

Comment:
We shouldn't use `async: false`, see above. With `async: true` (default mode) this line would always return undefined. Refactor the code so it still works with `async: true`.